### ISSUE TITLE
Problems with AAF adapters under Python 2.7 on Windows

### ISSUE THREAD

**Issue by flavienliger** [PixarAnimationStudios/OpenTimelineIO#316]
Hi,
First of all thanks for this nice library!
I've encountered several problems with the AAF adapter.

OS: Windows 10
Python: 2.7.14
OpenTimelineIO: Master
Pyaaf: 1.0

## First error

```Batch
otioview opentimelineio_contrib\adapters\tests\sample_data\trims.aaf

Traceback (most recent call last):
  File "C:\Python27\Scripts\otioview-script.py", line 11, in <module>
    load_entry_point('OpenTimelineIO==0.9.0.dev1', 'console_scripts', 'otioview')()
  File "c:\python27\lib\site-packages\opentimelineview\console.py", line 215, in main
    window.load(args.input)
  File "c:\python27\lib\site-packages\opentimelineview\console.py", line 162, in load
    self.timeline_widget.set_timeline(file_contents)
  File "c:\python27\lib\site-packages\opentimelineview\timeline_widget.py", line 743, in set_timeline
    self.add_stack(timeline.tracks)
  File "c:\python27\lib\site-packages\opentimelineview\timeline_widget.py", line 761, in add_stack
    new_stack = CompositionView(stack, parent=self)
  File "c:\python27\lib\site-packages\opentimelineview\timeline_widget.py", line 508, in __init__
    self.setScene(CompositionWidget(stack, parent=self))
  File "c:\python27\lib\site-packages\opentimelineview\timeline_widget.py", line 359, in __init__
    self._adjust_scene_size()
  File "c:\python27\lib\site-packages\opentimelineview\timeline_widget.py", line 365, in _adjust_scene_size
    scene_range = self.composition.trimmed_range()
  File "c:\python27\lib\site-packages\opentimelineio\core\item.py", line 102, in trimmed_range
    return self.available_range()
  File "c:\python27\lib\site-packages\opentimelineio\schema\stack.py", line 79, in available_range
    duration = max(child.duration() for child in self)
  File "c:\python27\lib\site-packages\opentimelineio\schema\stack.py", line 79, in <genexpr>
    duration = max(child.duration() for child in self)
  File "c:\python27\lib\site-packages\opentimelineio\core\item.py", line 89, in duration
    return self.trimmed_range().duration
  File "c:\python27\lib\site-packages\opentimelineio\core\item.py", line 102, in trimmed_range
    return self.available_range()
  File "c:\python27\lib\site-packages\opentimelineio\schema\track.py", line 128, in available_range
    opentime.RationalTime()
  File "c:\python27\lib\site-packages\opentimelineio\opentime.py", line 157, in __add__
    return RationalTime(self.value + other.value, self.rate)
TypeError: unsupported operand type(s) for +: 'float' and 'str'
```

## Second error

```Batch
otioview opentimelineio_contrib\adapters\tests\sample_data\misc_speed_effects.aaf

Traceback (most recent call last):
  File "C:\Python27\Scripts\otioview-script.py", line 11, in <module>
    load_entry_point('OpenTimelineIO==0.9.0.dev1', 'console_scripts', 'otioview')()
  File "c:\python27\lib\site-packages\opentimelineview\console.py", line 215, in main
    window.load(args.input)
  File "c:\python27\lib\site-packages\opentimelineview\console.py", line 158, in load
    **self.adapter_argument_map
  File "c:\python27\lib\site-packages\opentimelineio\adapters\__init__.py", line 141, in read_from_file
    **adapter_argument_map
  File "c:\python27\lib\site-packages\opentimelineio\adapters\adapter.py", line 133, in read_from_file
    **adapter_argument_map
  File "c:\python27\lib\site-packages\opentimelineio\plugins\python_plugin.py", line 128, in _execute_function
    return (getattr(self.module(), func_name)(**kwargs))
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 733, in read_from_file
    result = _transcribe(storage, masterMobs=masterMobs)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 171, in _transcribe
    child = _transcribe(mob, parent=item)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 184, in _transcribe
    child = _transcribe(slot, parent=item, masterMobs=masterMobs)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 268, in _transcribe
    child = _transcribe(item.segment, parent=item, masterMobs=masterMobs)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 398, in _transcribe
    result
opentimelineio.exceptions.OTIOError: Wrong duration? 8 should be 8 in Clip("Untitled SourceClip", MissingReference(None, None, {}), TimeRange(RationalTime(0, 24), RationalTime(8, 24)), {'AAF': {'Name': 'Untitled SourceClip', 'SourceID': 'urn:smpte:umid:060a2b34.01010101.01010f00.13000000.060e2b34.7f7f2a80.5b11c001.ad830f5c', 'DataDefinition': {u'Identification': 'urn:uuid:01030202-0100-0000-060e-2b3404010101', u'Name': u'Picture', u'Description': u'Picture Essence'}, 'ClassName': u'SourceClip', 'Length': '8', 'SourceMobSlotID': '1', 'StartTime': '0'}})
```

## Third error

Unfortunately I can't share my AAF files (they come from Avid), but have you ever encountered this error? it comes from my files? or a problem with pyaaf?

```batch
Traceback (most recent call last):
  File "C:\Python27\Scripts\otioview-script.py", line 11, in <module>
    load_entry_point('OpenTimelineIO==0.9.0.dev1', 'console_scripts', 'otioview')()
  File "c:\python27\lib\site-packages\opentimelineview\console.py", line 215, in main
    window.load(args.input)
  File "c:\python27\lib\site-packages\opentimelineview\console.py", line 158, in load
    **self.adapter_argument_map
  File "c:\python27\lib\site-packages\opentimelineio\adapters\__init__.py", line 141, in read_from_file
    **adapter_argument_map
  File "c:\python27\lib\site-packages\opentimelineio\adapters\adapter.py", line 133, in read_from_file
    **adapter_argument_map
  File "c:\python27\lib\site-packages\opentimelineio\plugins\python_plugin.py", line 128, in _execute_function
    return (getattr(self.module(), func_name)(**kwargs))
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 733, in read_from_file
    result = _transcribe(storage, masterMobs=masterMobs)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 171, in _transcribe
    child = _transcribe(mob, parent=item)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 184, in _transcribe
    child = _transcribe(slot, parent=item, masterMobs=masterMobs)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 268, in _transcribe
    child = _transcribe(item.segment, parent=item, masterMobs=masterMobs)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 257, in _transcribe
    child = _transcribe(component, parent=item, masterMobs=masterMobs)
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 203, in _transcribe
    and _extract_start_timecode(mobs[-1])
  File "c:\python27\lib\site-packages\opentimelineio_contrib\adapters\advanced_authoring_format.py", line 113, in _extract_start_timecode
    if s.segment.media_kind == 'Timecode'
  File "aaf\base.pyx", line 96, in aaf.base.AAFObject.__getitem__ (aaf\base.cpp:3975)
KeyError: 'Key Start not found'
```

Thank for you time.
Regards.

**Comment by jminor** [comment]
Yes, something is broken. It seems related to #301 but I'm confused about why the unit tests passed for that PR, when this is clearly broken. When I run the tests locally they fail with similar errors.

We'll dig a bit deeper, but in the meantime you can try using an earlier release: https://github.com/PixarAnimationStudios/OpenTimelineIO/releases

Sorry for the hassle and thanks for reporting the problem :)

**Comment by jminor** [comment]
So I thought I was seeing the same problem on my machine, but it turned out that I just had an old version of PyAAF installed, and was seeing a similar looking error message. With a fresh install of PyAAF, it seems to be working fine.

Specifically, this works for me:
```
% virtualenv /tmp/otio
% source /tmp/otio/bin/activate.csh
% pip install PySide2
% pip install ~/Downloads/PyAAF-1.0.0-cp27-none-macosx_10_6_intel.whl
% python setup.py install
% otioview opentimelineio_contrib/adapters/tests/sample_data/trims.aaf
% otioview opentimelineio_contrib/adapters/tests/sample_data/misc_speed_effects.aaf
```

@flavienliger, can you run `make test` and see if everything other than the AAF adapter is working for you? From the errors, it looks like a PyAAF mismatch somehow...

The only other difference I see is that we're mostly testing on Linux and Mac, but I believe it has been tested on Windows as well. I can double-check that.

**Comment by flavienliger** [comment]
> can you run make test and see if everything other than the AAF adapter is working for you? From the errors, it looks like a PyAAF mismatch 

@jminor, Yes the tests went well, only the AAF adapter seems to be a problem. 
Maybe it the PyAAF library for Windows and Python 2.7 which is the problem.

I did the test again in Python 3.6 and everything works.

**Comment by jminor** [comment]
I was able to reproduce the problem with Python 2.7 on Windows 10 - not 100% the same error, but I do see type mismatches between int/string when pulling values from an AAF.

@flavienliger, if you can work around this by using Python 3.6 in the short term until we get someone with a proper Windows dev setup who can fix this, that would be great.

**Comment by flavienliger** [comment]
For my "Third error" it is another problem, my AAF does not contain a Start value on some segments, which generates an error (it contains instead a "Components" list which itself contains a Start).
An example of quickfix https://github.com/flavienliger/OpenTimelineIO/commit/7ceec8ffd06e5f7c27d6a0e53c558eaaa8d136d4 .

@jminor thanks for your time, I use Python 3.6 so it doesn't bother me. It was to keep you informed.

**Comment by jminor** [comment]
Thanks for the bug fix. I know it is just a small fix, but could you follow the steps in our contributor guide? https://github.com/PixarAnimationStudios/OpenTimelineIO/wiki/Contributing Our company wants us to be strict about the process for accepting code contributions.

If that's too much hassle, maybe you could make a sample AAF (without any confidential material) that reproduces the bug and we can fix it.

**Comment by jminor** [comment]
@flavienliger are you able to submit a CLA and a pull request for your fix?

**Comment by flavienliger** [comment]
@jminor Yes, I just did it (send an email). Sorry for be late.

**Comment by ssteinbach** [comment]
@flavienliger Hey, when you get the chance, can you make a PR, thanks!  I'm bumping this to the next milestone for now.

**Comment by flavienliger** [comment]
Done [#362](https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/362).
(I was waiting for the rights, but no notifications warned me I already can :smiley:).

**Comment by ssteinbach** [comment]
Merged with #362
